src/main/java/org/nuxeo/ecm/core/storage/marklogic/MarkLogicQueryBuilder.java

```diff
@@ -1262,36 +1265,74 @@ public String getCountQuery(int countUpTo) {
         }
 
         private String createSearchQuery() {
-            return String.format("cts:search(fn:doc(),%s)", ctsQuery);
+            if (orderByClause == null) {
+                return String.format("cts:search(fn:doc(),%s)", ctsQuery);
```

Comment:
For low-level stuff prefer string concatenation instead of `String.format` which has a large parsing overhead.